python/fledge/services/core/server.py

```diff
@@ -1623,3 +1678,120 @@ async def add_audit(cls, request):
             raise web.HTTPInternalServerError(reason=ex)
 
         return web.json_response(message)
+
+    @classmethod
+    async def verify_token(cls, request):
+        """ Endpoint for verifycation of service bearer token received at registration time
+
+        :Example:
+            curl -H 'Authorization: Bearer evZGdrdmV.4dWFsY2dsaHVyZ.mFxdmdybXB5dXduaXJvc3g='
+            -X POST http://localhost:<core mngmt port>/fledge/service/verity_token
+
+        Authorization header must contain the Bearer token to verify
+        No post data
+        """
+
+        authData = request.headers.get('Authorization', "")
+        parts = authData.split("Bearer ")
+        if len(parts) != 2:
+            msg = "bearer token is missing"
+            raise web.HTTPBadRequest(reason=msg, body=json.dumps({"error": msg}))
+
+        bearer_token = parts[1]
+
+        if bearer_token is not None:
+            # Check input token exists in system
+            foundToken = False
+            services_list = ServiceRegistry.all()
+            for service in services_list:
+                t = service._bearer_token if service._bearer_token is not None else ""
+                if t == bearer_token:
+                    foundToken = True
+                    break
+            # Raise error if token does not exists
+            if foundToken == False:
+                msg = 'token does not exist in system'
+                raise web.HTTPBadRequest(reason=msg, body=json.dumps({"error": msg}))
```

Comment:
We should avoid looping, and should have a private dict for service name and jwt tokens (same way we have for single use tokens) and first,
decode the token, if decoded okay, find sub/aud and allow to proceed if matching that pair.

Reply:
I think this internal change could be postponed

Reply:
Having token attribute in slots as ServiceRecord member is not a good idea. As I stated earlier (in original commits PR), we are at risk of exposing tokens, having a better control is important. 
We must avoid from `def __repr__` and can see possible risks which will need explicit needs to delattr before sharing service entries among app components.

Reply:
@praveen-garg needed changes have been applied